SamTebbs33/pluto
# Release/v0.1

Miscellaneous:
* Update code to work with zig master
* Fix T() style casts
* No more assembly
* Improve readme
* Update to zig master and simplify build script
* Refactor compilation of architecture-specific code
* Separate arch-specific and arch-agnostic code
* Start using the zig build system

Testing:
* Added run time tests for PIT
* Add formatting check to CI
* Added unit and runtime tests
* Added unit and runtime tests
* Add x86 paging runtime tests
* Added unit tests for PIC
* Added runtime tests for TTY
* Added runtime tests for VGA
* Migrate to github actions for CI
* Import syscall tests into x86/arch.zig
* Only test a single build mode
* Added IDT unit and run-time tests
* Added gdt unit and runtime tests
* Add continuous integration with Azure
* Add runtime testing harness

Bug fixes:
* Fix some readme issues
* Stop rt-test script from hanging
* Remove debug option
* Fix gdb symbol-file argument
* Fixed testing

Features:
* Add simple stacktrace logging
* Load grub modules on mem init
* Add log tests
* Add serial tests
* Add 4kb paging
* Add x86 syscalls
* Add infrastructure for build options
* Add arch.registerInterruptHandler
* Add x86 paging
* Added PIT interface
* Add logging
* Add serial output
* Added the PIC interface
* Added GDT, IDT, IRQ, updated build.zig
* Add TTY and VGA interface
* Pass multiboot magic and header to kmain
* Add higher half support in linker script
* Add multiboot setup code
* Bootloader
